modules/ROOT/pages/cache-scope-invalidate.adoc

```diff
@@ -0,0 +1,107 @@
+include::_attributes.adoc[]
+endif::[]
+
+You can invalidate a cache key or a complete cache by using the Invalidate Key and Invalidate Cache components respectively. Alternatively, you can configure the caching strategy to invalidate entries automatically based on time.
+Invalidate cache entries when you want the Cache Scope to recalculate the value for those cache entries.
+
+== Invalidate a Cache Key
+
+Use the Invalidate Key component to invalidate the specified cache key in the referenced caching strategy.
+
+=== Configure the Invalidate Key Component
+
+To invalidate a cache key:
+
+. In *Anypoint Studio*, drag the *Invalidate Key* component to your flow.
+. Open the *Invalidate Key* properties.
+. Specify the *Caching Strategy* where the cache key to invalidate is stored.
+. Optionally, select the method for generating the cache key.
+
+=== Example Invalidate Key Configuration
+
+[source, xml, linenums]
+----
+<!-- Caching strategy configuration -->
+<ee:object-store-caching-strategy name="myCachingStrategy" keyGenerationExpression="#[payload.key]" />
+
+<flow name="cacheFlow">
+  <ee:cache cachingStrategy-ref="myCachingStrategy">
+      <!-- Processing logic inside the Cache Scope -->
+  </ee:cache>
+</flow>
+
+<flow name="invalidateItemFlow">
+  <!-- Invalidate Key component configuration -->
+  <ee:invalidate-key cachingStrategy-ref="myCachingStrategy" keyGenerationExpression="#[payload.key]" />
+</flow>
+----
+
+== Invalidate a Complete Cache
+
+Use the Invalidate Cache component to invalidate the complete cache in the referenced caching strategy.
+
+=== Configure the Invalidate Cache Component
+
+To invalidate a complete cache:
+
+. In *Anypoint Studio*, drag the *Invalidate Cache* component to your flow.
+. Open the *Invalidate Cache* properties.
+. Specify the *Caching Strategy* to invalidate.
+
+=== Example Invalidate Cache Configuration
+
+[source, xml, linenums]
+----
+<!-- Caching strategy configuration -->
+<ee:object-store-caching-strategy name="myCachingStrategy" />
+
+<flow name="cacheFlow">
+  <ee:cache cachingStrategy-ref="myCachingStrategy">
+      <!-- Processing logic inside the Cache Scope -->
+  </ee:cache>
+</flow>
+
+<flow name="invalidateCompleteCacheFlow">
+  <!-- Invalidate Cache component configuration -->
+  <ee:invalidate-cache cachingStrategy-ref="myCachingStrategy" />
+</flow>
+----
+
+== Automatic Invalidation
+
+When you configure the caching strategy, you can configure a custom object store and set the time to live (`entryTtl`) value to make stored cache keys expire over time.
+
+=== Configure the Object Store in your Caching Strategy
+
+. Follow the steps in xref:cache-scope-strategy.adoc[Set Up a Caching Strategy]
+. When configuring the object store for the strategy, set a value for *Entry ttl*.
++
+This value represents the time after which the cache entry expires.
+. Complete all the necessary values for the object store configuration.
+
+=== Example Caching Strategy Configuration
```

Comment:
Isn't this an example configuration of automatic invalidation?

Reply:
Yeah, so for this topic reexamine the title hierarchy and whether it makes clear how the subtopics relate to the container topics. 
I wonder if it would be more clear with titles of Steps to Configure... and Example Configuration of...?
= Invalidate Cache Entries
== Invalidate a Cache Key
=== Configure the Invalidate Key Component 
=== Example Invalidate [Cache?] Key Configuration
== Invalidate a Complete Cache
=== Configure the Invalidate Cache Component
=== Example Invalidate [Complete?] Cache Configuration
== Automatic Invalidation
=== Configure the Object Store in Your Caching Strategy
=== Example Caching Strategy Configuration [Example Automatic Invalidation Configuration?]